File: packages/sdk-python/veltrix_sdk/resources/environments.py
# -*- coding: utf-8 -*-

from .base import BaseResource

class EnvironmentsResource(BaseResource):
    """Handles operations related to environments.

    Environments are deployment targets used by the pipeline engine (e.g.
    ``dev``, ``staging``, ``production``).

    Note:
        The base route (``/api/environments``) matches the Community Edition
        server surface. The method surface below follows standard REST
        conventions; verify method-level shapes against the server as the
        OSS API stabilizes.
    """
    RESOURCE_PATH = "environments"

    def list(self, **kwargs):
        """Retrieves all environments for the authenticated organization."""
        return self._list(**kwargs)

    def get(self, environment_id, **kwargs):
        """Retrieves a specific environment by its ID."""
        return self._get(resource_id=environment_id, **kwargs)

    def create(self, name, description=None, **kwargs):
        """Creates a new environment."""
        data = {"name": name, "description": description}
        data = {k: v for k, v in data.items() if v is not None}
        return self._create(data=data, **kwargs)

    def update(self, environment_id, name=None, description=None, **kwargs):
        """Updates an existing environment."""
        data = {"name": name, "description": description}
        data = {k: v for k, v in data.items() if v is not None}
        return self._update(resource_id=environment_id, data=data, **kwargs)

    def delete(self, environment_id, **kwargs):
        """Deletes an environment by its ID."""
        return self._delete(resource_id=environment_id, **kwargs)

File: packages/sdk-js/src/resources/environments.ts
import { BaseResource } from './base';
import { AxiosRequestConfig } from 'axios';

// Deployment environments used by the pipeline engine (e.g. dev / staging /
// production).
//
// Provisional: the base route (/api/environments) matches the Community
// Edition server surface, but the method surface below follows standard REST
// conventions and may be refined as the OSS API stabilizes.

interface Environment {
  id: string;
  name: string;
  description?: string | null;
  createdAt?: string;
  updatedAt?: string;
}

interface CreateEnvironmentPayload {
  name: string;
  description?: string;
}

interface UpdateEnvironmentPayload {
  name?: string;
  description?: string;
}

export class EnvironmentsResource extends BaseResource {
  protected readonly RESOURCE_PATH = 'environments';

  async list(config?: AxiosRequestConfig): Promise<Environment[]> {
    return this._list<Environment[]>(undefined, config);
  }

  async get(environmentId: string, config?: AxiosRequestConfig): Promise<Environment> {
    return this._get<Environment>(environmentId, undefined, config);
  }

  async create(payload: CreateEnvironmentPayload, config?: AxiosRequestConfig): Promise<Environment> {
    const cleanedPayload = Object.entries(payload)
      .filter(([, value]) => value !== undefined)
      .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
    return this._create<Environment>(cleanedPayload, undefined, config);
  }

  async update(environmentId: string, payload: UpdateEnvironmentPayload, config?: AxiosRequestConfig): Promise<Environment> {
    const cleanedPayload = Object.entries(payload)
      .filter(([, value]) => value !== undefined)
      .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
    return this._update<Environment>(environmentId, cleanedPayload, undefined, config);
  }

  async delete(environmentId: string, config?: AxiosRequestConfig): Promise<Environment | null> {
    return this._delete<Environment>(environmentId, undefined, config);
  }
}
